### DISCUSSION TITLE
Does jsPsych provide plugin specifically tackle image processing issues?

### DISCUSSION THREAD

**Discussion by ontinglo** [jspsych/jsPsych#964] [Q&A] [answered]
dear mates

i am preparing an experiment using jsPsych, during the experiment i would show human face stimuli. I could cut all non-facial features in these face pictures such as hairstyle (so the face stimuli would become the shape of ellipse). In my practice of running relevant experiments in Matlab or PsychoPy, I would add a raised cosine filter (or any other filters) to the face stimuli to make sure that the edge between the face stimuli and the background will be smooth. 

does jsPsych provide plugin to deal with such image processing issues? 
such image filtering would be done real-time and in trial by trial sequence, would jsPsych be powerful enough to accomplish such tasks?

thanks a lot！

**Comment by jodeleeuw** [comment, marked as answer]
Hi @ontinglo ,

There's no integrated image processing in jsPsych. But you should be able to use any image processing you'd like that's available through CSS or JavaScript. 

Some examples:
* [CSS filters](https://css-tricks.com/almanac/properties/f/filter/)
* [CamanJS Image Processing Library](http://camanjs.com/)
* [Lena.JS Image Processing Library](https://github.com/davidsonfellipe/lena.js/)

> such image filtering would be done real-time and in trial by trial sequence, would jsPsych be powerful enough to accomplish such tasks?

This depends on several factors. JavaScript is running on the participant's computer, so a lot depends on the particular device that participant is using. It'll also depend on how large the images are that you want to show and how complex the filtering operation is. If the compute time for the filter operation is under, say, 8ms or so then you should be fine, since screens are refreshing every 16ms on most devices. I suspect that it'll be fine, unless you are doing filtering that is particularly processing heavy.

**Reply by ontinglo** [reply]
Thank you very much Josh 

and thank you (and your team) for building such wonderful tool for us to move our experiments online! Salute!
